api/scanner/process_photo.go

```diff
@@ -96,10 +96,72 @@ func processPhoto(tx *sql.Tx, imageData *EncodeMediaData, photoCachePath *string
 		return false, errors.Wrap(err, "error processing photo highres")
 	}
 
-	// Generate high res jpeg
 	var photoDimensions *PhotoDimensions
 	var baseImagePath string = photo.Path
 
+	// Save original photo to database
+	if origURL == nil {
+		didProcess = true
+
+		// Make sure photo dimensions is set
+		if photoDimensions == nil {
+			photoDimensions, err = GetPhotoDimensions(baseImagePath)
+			if err != nil {
+				return false, err
+			}
+		}
+
+		if err = saveOriginalPhotoToDB(tx, photo, imageData, photoDimensions); err != nil {
+			return false, errors.Wrap(err, "saving original photo to database")
+		}
+	}
+
+	mediaType, err := getMediaType(photo.Path)
+	if err != nil {
+		return false, errors.Wrap(err, "could determine if media was photo or video")
+	}
+
+	if mediaType.isRaw() {
+		sideCarPath, err := scanForSideCarFile(photo.Path)
+		if err == nil {
+			currentFileHash := hashSideCarFile(sideCarPath)
+			if photo.SideCarHash != currentFileHash {
+				fmt.Printf("Detected changed sidecar file for %s recreating JPG's to reflect changes\n", photo.Path)
+
+				// update high res image may be cropped so dimentions and file size can change
+				baseImagePath := path.Join(*photoCachePath, highResURL.MediaName)  // update base image path for thumbnail
+				tempHighResPath := baseImagePath + ".hold"
+				os.Rename(baseImagePath, tempHighResPath)
+				err = generateSaveHighResJPEG(tx, photo.MediaID, imageData, highResURL.MediaName, baseImagePath, highResURL.UrlID)
+				if err != nil {
+					os.Rename(tempHighResPath, baseImagePath)
+					return false, errors.Wrap(err, "recreating high-res cached image")
+				}
+				os.Remove(tempHighResPath)
+
+				// update thumbnail image may be cropped so dimentions and file size can change
+				thumbPath := path.Join(*photoCachePath, thumbURL.MediaName)
+				tempThumbPath := thumbPath + ".hold" // hold onto the original image incase for some reason we fail to recreate one with the new settings
+				os.Rename(thumbPath, tempThumbPath)
+				err := generateSaveThumbnailJPEG(tx, photo.MediaID, thumbURL.MediaName, photoCachePath, baseImagePath, thumbURL.UrlID)
+				if err != nil {
+					os.Rename(tempThumbPath, thumbPath)
+					return false, errors.Wrap(err, "recreating thumbnail cached image")
+				}
+				os.Remove(tempThumbPath)
+
+
+
+				// save new side car hash
+				_, err = tx.Exec("UPDATE media SET side_car_hash = ? WHERE media_id = ?", currentFileHash, photo.MediaID)
+				if err != nil {
+					return false, errors.Wrapf(err, "could not update side car hash (%d, %s)", photo.MediaID, currentFileHash)
+				}
+			}
+		}
```

Comment:
Maybe move all this (everything inside of `if mediaType.isRaw() { ... }` to a new function.

Reply:
moved :) this function grew slowly so it's size snuck up on me haha